rllib/models/tf/visionnet.py

```diff
@@ -35,10 +38,11 @@ def __init__(self, obs_space, action_space, num_outputs, model_config,
                 padding="same",
                 data_format="channels_last",
                 name="conv{}".format(i))(last_layer)
+
         out_size, kernel, stride = filters[-1]
 
         # No final linear: Last layer is a Conv2D and uses num_outputs.
-        if no_final_linear:
+        if no_final_linear and num_outputs:
```

Comment:
When do we ever set this to None? I don't see it in the modelv2 docs.

Reply:
I'll add this to the docs.

Leaving it None (or 0) simply means, we don't care about the `num_outputs`, the Model will figure this out by itself, will not generate a logits layer or last layer and just set its `self.num_outputs` to whatever that'll be (after creating the underlying model).

This has been around for some time in fcnets (see SAC model when not using a Conv2D state preprocessor). Just added it to conv2d nets as well to make them behave the same.
With this in place, we can get rid of the artificially hard-coding num_outputs=256 in case of SAC image space (Conv2D).
```
    if config["use_state_preprocessor"]:
        num_outputs = 256  # Flatten last Conv2D to this many nodes.
    else:
        num_outputs = 0
```

Without this option, when you look at the `make_model` for DQN in dqn_torch/tf_policy, we need to somehow infer the last layer size from the given config (fcnet_hiddens), which is unwieldy (we use 256 hard-coded!). When we create custom models, we usually wouldn't know, what the underlying default model will be (after all, this is the exact reason, we use ModelCatalog to figure this out for us).